src/main/scala/diplomaticobjectmodel/DiplomaticObjectModelUtils.scala

```diff
@@ -186,4 +186,34 @@ object DiplomaticObjectModelAddressing {
         writeMaskGranularity = granWidth
       )
     }
+
+  private def getInterruptNumber(r: ResourceValue): BigInt = {
+    r match {
+      case ResourceInt(value: BigInt) => value
+      case _ => throw new IllegalArgumentException
+    }
+  }
+
+  private def getDeviceName(device: Device): String = {
+    device match {
+      case sd:SimpleDevice => sd.asInstanceOf[SimpleDevice].devNamePlusAddress
+      case _ => throw new IllegalArgumentException
+    }
+  }
+
+  def describeInterrupts(name: String, resources: ResourceBindings): Seq[OMInterrupt] = {
+    val int = resources("int")
+    int.flatMap {
+      b =>
+        val grandParentOpt = b.device.get.parent
+        grandParentOpt.map {
+         gp =>
```

Comment:
This is another instance where using a `for` comprehension may improve readability by reducing the level of nesting required:

```scala
for {
  b <- int
  grandParentOpt = b.device.get.parent
  gp <- grandParentOpt
} yield OMInterrupt(...)
```

Reply:
This coding style should be brought up at the engineering forum

Reply:
changed